### PR TITLE
Modify CI to capture dumps on hanging CI builds

### PR DESCRIPTION
**Pull request by javiercn**

### CODE REVIEW

File: runWithProcDump.ps1 [hunk 1]
```diff
@@ -0,0 +1,30 @@
+try
+{
+  $procDumpFolder = "$PSScriptRoot/obj";
+  mkdir "$procDumpFolder" -Force;
+  Invoke-WebRequest https://download.sysinternals.com/files/Procdump.zip -OutFile "$procDumpFolder/procdump.zip";
+  Expand-Archive "$procDumpFolder/procdump.zip" -DestinationPath "$procDumpFolder" -Force;
+
+  $job = Start-Job {    
+    # $sleepTime = (1 * 20 * 60)
```
Comment (pranavkm) [inline runWithProcDump.ps1]: Remove?

File: .vsts-pipelines/builds/ci-public.yml [hunk 2]
```diff
@@ -2,14 +2,5 @@ trigger:
...
-  repositories:
-  - repository: buildtools
-    type: github
-    endpoint: DotNet-Bot GitHub Connection
-    name: aspnet/BuildTools
-    ref: refs/heads/master
-
 phases:
-- template: .vsts-pipelines/templates/project-ci.yml@buildtools
+- template: ../../.vsts-pipelines/templates/project-ci.yml
```
[the remark this answers is not in the document]
Comment (javiercn) [reply .vsts-pipelines/builds/ci-public.yml]: Sure, this is just temporary

File: runWithProcDump.ps1 [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    $processes = Get-Process dotnet*;
+    $processes | Format-Table;
+    Write-Host "$procDumpFolder/procdump.exe";
+    
+    $processes |
+     Select-Object -ExpandProperty ID | 
+     ForEach-Object { &"$procDumpFolder/procdump.exe" -accepteula -ma $_ $dumpsFolder }
+  }
+  ./run.ps1 default-build $args
+  Stop-Job $job
```
[the remark this answers is not in the document]
Comment (javiercn) [reply runWithProcDump.ps1]: The job is tied to the session, you can think of it as a "spawning a thread"

File: .vsts-pipelines/templates/phases/default-build.yml [hunk 4]
```diff
@@ -0,0 +1,118 @@
+#
+# Obsolete! New projects should use .azure/templates/jobs/default-build.yml instead
```
[the remark this answers is not in the document]
Comment (javiercn) [reply .vsts-pipelines/templates/phases/default-build.yml]: This is what @natemcmaster told me to do and is using the existing template. This is just until we capture dumps

File: runWithProcDump.ps1 (hunk 3 above)
[the remark this answers is not in the document]
Comment (ryanbrandenburg) [reply runWithProcDump.ps1]: Ah, if it's tied to this process then that's fine. I was thinking of it as a process instead of a thread.

**Pull request merged**